src/server/game/Entities/Creature/Creature.h

```diff
@@ -659,7 +659,7 @@ class Creature : public Unit, public GridObject<Creature>, public MapObject
 
         // Handling caster facing during spellcast
         void SetTarget(ObjectGuid guid) override;
-        void FocusTarget(Spell const* focusSpell, WorldObject const* target);
+        void FocusTarget(Spell const* focusSpell, ObjectGuid targetGuid);
```

Comment:
the wotlk one is (should be) handled by SPELL_ATTR5_DONT_TURN_DURING_CAST

> Should I just remove the creatures target then ?

don't call FocusTarget if caster == target like it is used in other places